tls/s2n_connection.c

```diff
@@ -1537,3 +1539,18 @@ int s2n_connection_get_selected_client_cert_signature_algorithm(struct s2n_conne
 
     return S2N_SUCCESS;
 }
+
+/*
+ * Internal API.
+ *
+ * Gets the config set on the connection.
+ */
+int s2n_connection_get_config(struct s2n_connection *conn, struct s2n_config **config) {
+    if (s2n_fetch_default_config() == conn->config) {
```

Comment:
Please make sure you check for nulls before dereferencing, particularly with "public" methods.
```suggestion
int s2n_connection_get_config(struct s2n_connection *conn, struct s2n_config **config) {
    POSIX_ENSURE_REF(conn);
    POSIX_ENSURE_REF(config);
    if (s2n_fetch_default_config() == conn->config) {
```